### PR TITLE
ETCD-460: add basic backup e2e tests

### PR DESCRIPTION
**Pull request by tjungblu**

### CODE REVIEW

File: test/e2e/backup_test.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+				klog.Infof("error while getting job list: %v", err)
+				return false, nil
+			}
+
+			// the cronjob will keep at most 5 successful runs, we're waiting until the list has that many jobs completed
+			succeededJobs := 0
+			for _, j := range jobList.Items {
+				ownedByOurCron := false
+				for _, reference := range j.OwnerReferences {
+					if reference.Kind == "CronJob" && reference.Name == backupCrd.Name {
```
Comment (Elbehery) [inline test/e2e/backup_test.go:111]: nit: can we replace the literal string "CronJob" with a const or similar ?

File: test/e2e/backup_test.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+}
+
+func ensureHostPathPVC(t *testing.T, pvcName string) {
+	kubeConfig, err := framework.NewClientConfigForTest("")
+	require.NoError(t, err)
+
+	c, err := corev1client.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+
+	pv := v1.PersistentVolume{
```
Comment (Elbehery) [inline test/e2e/backup_test.go:174]: I cant get why we create the PV manually, isn't PVC creates a PV automatically as long as there is `StorageClass` ?

Comment (tjungblu) [reply test/e2e/backup_test.go:174]: dynamic provisioning doesn't work with hostPaths IIRC

Comment (Elbehery) [reply test/e2e/backup_test.go:174]: really :O .. how the hostpath CSI driver works !! .. let me check

Comment (tjungblu) [reply test/e2e/backup_test.go:179]: is there a CSI driver? I thought that's still hardcoded somewhere in kubelet

Comment (Elbehery) [reply test/e2e/backup_test.go:179]: yes u r right, it does not 👍🏽

File: test/e2e/backup_test.go (hunk 1 above)
Comment (Elbehery) [reply test/e2e/backup_test.go]: thanks

File: test/e2e/backup_test.go (hunk 2 above)
Comment (tjungblu) [reply test/e2e/backup_test.go:179]: hahaha great, well it's a demo :)

**Comment by tjungblu** [comment]
/hold

I'm going to split the commits into logic fixes and the tests once we have the full loop working

**Pull request merged**